fq_default/ctx_modulus.c

```diff
@@ -0,0 +1,48 @@
+/*
+    Copyright (C) 2021 William Hart
+
+    This file is part of FLINT.
+
+    FLINT is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.  See <https://www.gnu.org/licenses/>.
+*/
+
+#include "fq_default.h"
+
+void fq_default_ctx_modulus(fmpz_mod_poly_t p, const fq_default_ctx_t ctx)
+{
+   if (ctx->type == 1)
+   {
+      nmod_poly_t mod;
+      nmod_poly_init(mod, fmpz_get_ui(fq_zech_ctx_prime(ctx->ctx.fq_zech)));
+      nmod_poly_set(mod, fq_zech_ctx_modulus(ctx->ctx.fq_zech));
+      fmpz_mod_poly_set_nmod_poly(p, mod);
```

Comment:
`fq_zech_ctx_modulus(ctx->ctx.fq_zech)` is already an `nmod_poly`, so no need for this extra copy `mod`?